File: solutions_5652388522229760_1/java/Desliner/ACountingSheep.java
package codejam.y2016.qualification;

import lib.FastScanner;

import java.io.FileOutputStream;
import java.io.PrintStream;
import java.util.HashSet;
import java.util.Set;

/**
 * @author Max Myslyvtsev
 * @since 4/8/16
 */
public class ACountingSheep {

    public static void main(String[] args) throws Exception {
        //FastScanner in = new FastScanner("/Users/max/Work/Source/foobar/src/main/resources/input.txt");
        FastScanner in = new FastScanner("/Users/max/Downloads/A-large.in");
        PrintStream out = new PrintStream(new FileOutputStream("/Users/max/Work/Source/foobar/src/main/resources/output.txt"));
        int T = in.nextInt();
        for (int t = 1; t <= T; t++) {
            int N = in.nextInt();
            Long a = solve(N);
            String answ = "Case #" + t + ": " + (a != null ? a.toString() : "INSOMNIA");
            out.println(answ);
            System.out.println(answ);
        }
    }

    private static Long solve(long N) {
        if (N == 0) {
            return null;
        }
        Set<Character> digits = new HashSet<>();
        for (int i = 1; ; i++) {
            long q = N * i;
            String s = String.valueOf(q);
            for (char ch : s.toCharArray()) {
                digits.add(ch);
            }
            if (digits.size() == 10) {
                return q;
            }
        }
    }
}

File: solutions_5652388522229760_0/Python/Chiffre/A Counting Sheep.py
def removal(n,d):
    for j in map(int,list(str(n))):
        if j in d:
            d.remove(j)
    return d

t=input()
l=[]
for i in range(t):
    l.append(input())
for i in range(len(l)):
    n=l[i]
    c=n
    if c==0:
        print 'Case #'+str(i+1)+': '+'INSOMNIA'
    else:
        d=range(10)
        while n/c<200:
            if not d:
                break
            else:
                removal(n,d)
                n=n+c
        if d:
            print 'Case #'+str(i+1)+': '+'INSOMNIA'
        else:
            print 'Case #'+str(i+1)+': '+str(n-c)        
